package.json

```diff
@@ -26,20 +26,23 @@
     "npm": "6.4.1"
   },
   "dependencies": {
-    "@silvermine/eslint-plugin-silvermine": "2.2.1",
-    "@typescript-eslint/eslint-plugin": "1.2.0",
+    "@typescript-eslint/parser": "2.13.0",
     "typescript": "3.3.1"
   },
   "devDependencies": {
     "@commitlint/cli": "7.5.2",
     "@commitlint/travis-cli": "7.5.2",
+    "@silvermine/eslint-plugin-silvermine": "git+https://github.com/silvermine/eslint-plugin-silvermine.git#9a6e324f951fe89d5b6b115a5881cd6ff14b5f96",
+    "@typescript-eslint/eslint-plugin": "1.2.0",
     "cz-conventional-changelog": "2.1.0",
     "grunt": "1.0.3",
-    "grunt-eslint": "21.0.0",
+    "grunt-eslint": "22.0.0",
     "standard-version": "7.0.0"
   },
   "peerDependencies": {
-    "eslint": ">= 5.11.1"
+    "@silvermine/eslint-plugin-silvermine": "git+https://github.com/silvermine/eslint-plugin-silvermine.git#9a6e324f951fe89d5b6b115a5881cd6ff14b5f96",
+    "@typescript-eslint/eslint-plugin": "^2.13.0",
```

Comment:
Can we move these back up to the `dependencies` section? Is there a reason we have to have them in `peerDependencies` with eslint@6?

We're going to require a specific version of the plugins to be able to use the config anyway and putting them in the dependencies will prevent us from having to install them locally in every project.  

Also, we could then only have one reference to these packages. You could remove them from `devDependencies`.

Either way, please lock doen the `@typescript-eslint/eslint-plugin` package to the exact version (i.e. no `^`) please.

Reply:
Yes, that is a difference between the way plugins work in eslint 6 vs eslint 5.
> Previously, ESLint loaded plugins relative to the location of the ESLint package itself. As a result, we suggested that users with global ESLint installations should also install plugins globally, and users with local ESLint installations should install plugins locally. However, due to a design bug, this strategy caused ESLint to randomly fail to load plugins and shareable configs under certain circumstances, particularly when using package management tools like lerna and Yarn Plug n' Play.

> ESLint v6 resolves plugins relative to the end user's project by default, and always resolves shareable configs and parsers relative to the location of the config file that imports them.”

https://eslint.org/docs/user-guide/migrating-to-6.0.0#package-loading-simplification

What do you think of the above?

As for versioning, when a package is a dependency, locking to an exact version should be OK. But if we're specifying a package as a peer dependency, would it be practical lock to an exact version?

Reply:
I read the above as "Don't install plugins globally (i.e.`npm i -g my-plugin`)", not "Plugins need to be a direct dependency of the end user's project". Dependencies of a dependency should be resolvable from the project directory, just like `peerDependencies`. 

@jthomerson Do you have any input on this?

Reply:
I understand it the same was @jimjenkins5 does.

Also, our config is naturally married to a specific version of eslint. If we allow fuzzy-matching of the version of eslint, then we don't know the resulting behavior. Only by locking to a specific version will we be able to guarantee that the behavior of our rules is stable for a given version of the config. That means that eslint really is a dependency, not something we can simply rely on the user to provide some (possibly conflicting) version of.

Also, providing a specific version of eslint as a dependency of the config makes its usage much simpler for users - they simply depend on our config, and don't have to worry about marrying a compatible version of eslint to the config. This is a huge consideration - we're using this config in many dozens of repos, and having to go add a compatible version of eslint to all of them would be a giant pain, and an ongoing maintenance problem.

So, unless @fketchakeu has other reasons that it needs to be a peer dependency, I vote that it's back to a regular dependency.

Reply:
Concerning listing `ESLint` as a dependency, (please correct me if I'm wrong) I don't think that it is what @jpjenkins meant. The Config does not directly depend on eslint at all. Eslint has always been listed as a peer-dependency.

Concerning the two plugins, installing some dependencies of `@silvermine/eslint-config-silvermine` in the folder `@silvermine/eslint-config-silvermine/node_modules` is a possible behavior of npm. There is nothing that requires that those plugins should be installed in the end user project's `node_modules` folder as long as they are not listed in that end user project's package.json.

> ESLint v6 resolves plugins relative to the end user's project by default

So not listing the plugins as peer-dependencies of `@silvermine/eslint-config-silvermine` means that the user is relying on “phantom dependencies”. That is why I do not agree that the plugins should not be listed as peer-dependencies of `@silvermine/eslint-config-silvermine`. However, since @jpjenkins said (via another channel) that we have the same issue with previous versions of `@silvermine/eslint-config-silvermine`, and you own the project, I'm willing to acquiesce. Let me know if you still want to pursue the requested change.

Reply:
@jthomerson I will leave this decision to you. Here's a little more information:

   * [ESLint does recommend that sharable configs define their plugins as a `peerDependency`](https://eslint.org/docs/developer-guide/shareable-configs#publishing-a-shareable-config)
   * Using `npm link` or installing the config from the file system (i.e. `npm i ../eslint-plugin-silvermine` will result in ESLint failing to find the plugin.

The second issue is not new, as @fketchakeu mentioned. It also has only ever been an issue while developing the changes to the config.

Reply:
@fketchakeu you're right: I misread the diff and was talking a lot about `eslint` while you guys were talking about the plugins.

Still, after discussing at length with @jimjenkins5 IRL, we decided to keep the plugins listed as actual (not peer) dependencies. For posterity, here's the reasoning:

 1. As I noted above, our config is naturally married to a specific version of eslint and the associated plugins. If we allow fuzzy-matching of their versions, then we don't know the resulting behavior. Only by locking to a specific version (as much as possible) will we be able to guarantee that the behavior of our rules is stable for a given version of the config. Thus, we can't simply rely on the user to provide some (possibly conflicting) version of these dependencies.
 2. Also noted above (although with a slight edit), providing a specific version of ~eslint~ _these plugins_ as a dependency of the config makes its usage much simpler for users - they simply depend on our config, and don't have to worry about marrying ~a compatible version of eslint~ compatible versions of the plugins to the config. This is a huge consideration - we're using this config in many dozens of repos, and having to go add ~a compatible version of eslint~ _compatible versions of each plugin~ to all of them would be a giant pain, and an ongoing maintenance problem.
 3. Using a peer dependency requires that we also specify a dev dependency. That's another place where we have to keep things in sync. Versions there can easily get out of sync. Actually, in your pull request, you've done this yourself - the peer dependency for `@typescript-eslint/eslint-plugin` is defined as `^2.13.0`, but the dev dependency is using `1.2.0`. This is no big deal - we could fix it with a one-line change, but it's the perfect example of why we're trying to avoid putting this load on the user (previous bullet point).
 4. You're correct that `eslint` recommends using peer dependencies for shareable configs (@jimjenkins5 provided the link to that recommendation). However, the only case where we see that being a benefit over our approach of defining them as actual dependencies is in the case where you're `npm link` -ing to the config. Since that's really something that should only be done during development, that's a cost that only we developers must take on. This is - to us - much more acceptable than the cost that we'd impose on our users if we defined these as peer dependencies and they had to install compatible versions of them.

One additional note: `eslint` will remain a peer dependency. We would like to move it to a dependency for all the same reasons noted above, but that has caused major headaches in the past, especially with `grunt-eslint`. So, we leave it. For now, we'll leave it with the permissive `>=6.0.0` that you have in this PR, but eventually (as we get closer to cutting a release), we may lock that down a bit more for the same reasons as above (the ones that are applicable).

Hope that helps. Thanks for your willingness to acquiesce. Let us know if you'd like to fix up the PR for a re-review, or if you'd like @jimjenkins5 to finish it up.

Reply:
@jthomerson @jimjenkins5 I just pushed the changes.
Thanks.